Apps/Common/src/Services/EmailQueueService.cs

```diff
@@ -128,19 +128,20 @@ public void CloneAndQueue(Guid emailId, bool shouldCommit = true)
         }
 
         /// <inheritdoc />
-        public void QueueNewInviteEmail(string hdid, string toEmail, Uri activationHost)
+        public void QueueNewInviteEmail(string hdid, string toEmail, Uri activationHost, int verificationExpiryHours)
         {
             Dictionary<string, string> keyValues = new Dictionary<string, string>();
             MessagingVerification invite = new MessagingVerification();
             invite.InviteKey = Guid.NewGuid();
             invite.HdId = hdid;
-            invite.ExpireDate = DateTime.UtcNow.AddDays(VerificationExpiryDays);
+            invite.ExpireDate = DateTime.UtcNow.AddHours(verificationExpiryHours);
 
             string hostUrl = activationHost.ToString();
             hostUrl = hostUrl.Remove(hostUrl.Length - 1, 1); // Strips last slash
 
-            keyValues.Add(INVITE_KEY_VARIABLE, invite.InviteKey.ToString());
+            keyValues.Add(INVITE_KEY_VARIABLE, hdid);
```

Comment:
we dont want to send the hdid as a key to the user. We want the same invite key as the previous email if the email is the same

Reply:
@FuriousLlama I agree, I have missed that we need email and user as the key here. Do you think if we can use hash function to generate this invite key for this?

Reply:
I think the generated GUID  should be sufficient

Reply:
@FuriousLlama How do we retrieve the existing verification email for a user with a specific email? Do we need to query db by email and content template of this emails? 
Also, generating GUID requires us to query DB again, but using hash doesn't require a query to DB for this scenario.

Reply:
The EmailQueueService does not need to know the business logic of generating keys or expiring old ones. That is done on the UserProfileService and UserSmsService. Both already have the previous email that was sent. It just needs to check if it is the same as the new one and if it is pass the invite key

Reply:
per our team discussion, passing the existing inviteKey instead